runtime/nvqir/cutensornet/tensornet_state.h

```diff
@@ -19,6 +19,8 @@ class TensorNetState {
   std::size_t m_numQubits;
   cutensornetHandle_t m_cutnHandle;
   cutensornetState_t m_quantumState;
+  // Track id of tensors that are applied to the state tensors.
+  int64_t m_tensorId = -1;
```

Comment:
`-1` magic value. What does it mean?

Reply:
It's intended to represent an invalid index value. When applying gate tensors to the qubit state, `cutensornet` will return the index of the gate tensor. Previously, we just discarded this index, now I use it (stored in this var) as a way to tell if the underlying tensor network has been modified beyond an initial state.

Reply:
Sounds good.

I'd give the magic value a name so it can be found and understood though. Somewhere at the top of the file perhaps,
```c++
/// This is used to track whether the tensor state is default initialized vs initialized in some other state.
constexpr std::int64_t InvalidTensorIndexValue = -1;
```

Reply:
Fixed in https://github.com/NVIDIA/cuda-quantum/pull/1420/commits/bdce5e63d490551986f0191b6cc5fa45e943b527